go/codegen/experimental/codegen_step_run.go

```diff
@@ -0,0 +1,57 @@
+package experimental
+
+import (
+	"bytes"
+	"html/template"
+
+	"sqlflow.org/sqlflow/go/ir"
+	pb "sqlflow.org/sqlflow/go/proto"
+)
+
+type runStepFilter struct {
+	StepIndex  int
+	DataSource string
+	Select     string
+	ImageName  string
+	Parameters string
+	Into       string
+	Submitter  string
+}
+
+// GenerateRun returns the step code for TO RUN.
+func GenerateRun(runStmt *ir.RunStmt, stepIndex int, session *pb.Session) (string, error) {
+	ds, err := GeneratePyDbConnStr(session)
+	if err != nil {
+		return "", err
+	}
+
+	filter := &runStepFilter{
+		StepIndex:  stepIndex,
+		DataSource: ds,
+		Select:     runStmt.Select,
+		ImageName:  runStmt.ImageName,
+		Parameters: ir.AttrToPythonValue(runStmt.Parameters),
+		Into:       runStmt.Into,
+		Submitter:  getSubmitter(session),
+	}
+
+	var program bytes.Buffer
+	var runTemplate = template.Must(template.New("Run").Parse(runStepTemplate))
+	err = runTemplate.Execute(&program, filter)
+	if err != nil {
+		return "", err
+	}
+
+	return program.String(), nil
+}
+
+const runStepTemplate = `
+def step_entry_{{.StepIndex}}():
```

Comment:
What if `TO RUN` is not running a Python program? For example, a C/C++ executable?

Reply:
Good question. Currently we have the requirement that the runnable docker image is built based on sqlflow/sqlflow:step. It has runtime library inside this image which already contains python.